api/oci/ociutils/ref.go

```diff
@@ -0,0 +1,83 @@
+package ociutils
+
+import (
+	"strings"
+
+	"github.com/mandelsoft/goutils/generics"
+	"github.com/opencontainers/go-digest"
+)
+
+// ParseVersion parses the version part of an OCI reference consisting
+// of an optional tag and/or digest.
+func ParseVersion(vers string) (ArtVersion, error) {
+	if strings.HasPrefix(vers, "@") {
+		dig, err := digest.Parse(vers[1:])
+		if err != nil {
+			return ArtVersion{}, err
+		}
+		return ArtVersion{
+			Tag:    nil,
+			Digest: &dig,
+		}, nil
+	}
+
+	i := strings.Index(vers, "@")
+	if i > 0 {
+		dig, err := digest.Parse(vers[i+1:])
+		if err != nil {
+			return ArtVersion{}, err
+		}
+		return ArtVersion{
+			Tag:    generics.Pointer(vers[:i]),
+			Digest: &dig,
+		}, nil
+	}
+	return ArtVersion{
+		Tag: &vers,
+	}, nil
+}
+
+// ArtVersion is the version part of an OCI reference consisting of an
+// optional tag and/or digest. Both parts may be nil, if a reference
+// does not include a version part.
+type ArtVersion struct {
+	// +optional
+	Tag *string `json:"tag,omitempty"`
+	// +optional
+	Digest *digest.Digest `json:"digest,omitempty"`
+}
+
+func (v *ArtVersion) VersionSpec() string {
+	vers := ""
+	if v != nil {
+		if v.Tag != nil {
+			vers = *v.Tag
+		}
+		if v.Digest != nil {
+			vers += "@" + string(*v.Digest)
+		}
+		if vers == "" {
+			return "latest"
+		}
+	}
+	return vers
+}
```

Comment:
```suggestion
func (v *ArtVersion) VersionSpec() string {
	if v == nil {
	      return ""
	}
	
	var version string
      if v.Tag != nil {
		version = *v.Tag
	}
	if v.Digest != nil {
		version += "@" + string(*v.Digest)
	}
	// an initialized ArtVersion with nil fields is equivalent to latest
	if version == "" {
		version = "latest"
	}
	
	return version
}
```
just a nit, but easier to read due to only one layer of if

Reply:
done